x/concentrated-liquidity/genesis.go

```diff
@@ -184,16 +195,36 @@ func (k Keeper) ExportGenesis(ctx sdk.Context) *genesis.GenesisState {
 		if err != nil {
 			panic(err)
 		}
+
 		feeAccumPositionRecord, err := feeAccumObject.GetPosition(feePositionKey)
 		if err != nil {
 			panic(err)
 		}
 
-		positionData = append(positionData, genesis.PositionData{
-			LockId:         lockId,
-			Position:       &position,
-			FeeAccumRecord: feeAccumPositionRecord,
-		})
+		// Retrieve uptime incentive accumulator state for position
+		positionName := string(types.KeyPositionId(position.PositionId))
+		uptimeAccumulators, err := k.GetUptimeAccumulators(ctx, position.PoolId)
+		if err != nil {
+			panic(err)
+		}
+
+		var uptimeAccumRecords [][]accum.Record
+		for uptimeIndex := range types.SupportedUptimes {
+			accumRecord, err := uptimeAccumulators[uptimeIndex].GetPosition(positionName)
+			if err != nil {
+				panic(err)
+			}
+
+			uptimeAccumRecords[uptimeIndex] = append(uptimeAccumRecords[uptimeIndex], accumRecord)
+
+			positionData = append(positionData, genesis.PositionData{
+				LockId:            lockId,
+				Position:          &position,
+				FeeAccumRecord:    feeAccumPositionRecord,
+				UptimeAccumRecord: uptimeAccumRecords[uptimeIndex],
+			})
+		}
+
 	}
```

Comment:
This duplicates positions. We need to have one `PositionData` structure object associated with a position.

We don't need a `PositionData` for every uptime. This might introduce errors and waste genesis state for no reason.

Suggest following the matrix approach we discussed to preserve the relationship between an uptime accumulator and its positions while avoiding duplicating `PositionData`

Reply:
changed! lmk what you think